packages/@uppy/companion/src/server/logger.js

```diff
@@ -56,7 +56,7 @@ const log = ({ arg, tag = '', level, traceId = '', color = (message) => message
   function msgToString () {
     // We don't need to log stack trace on special errors that we ourselves have produced
     // (to reduce log noise)
-    if ((arg instanceof ProviderApiError || arg instanceof ProviderAuthError) && typeof arg.message === 'string') {
+    if ((arg instanceof ProviderApiError || arg instanceof ProviderUserError || arg instanceof ProviderAuthError) && typeof arg.message === 'string') {
```

Comment:
It should be noted that using `instanceof` is typically terrible for performance, we'd be better off check some property.

Reply:
any source for that?

Reply:
It's a very pleasant patterns for errors though and to please TS as well. Pretty sure the performance issue is negligible in the grand schemes of things when working with Node.js.

Reply:
My source is https://github.com/nodejs/node/pull/50874#issuecomment-1824044111, although I reckon that's more of an appeal to authority than an actual proof. Maybe we could try to make an intermediate parent class for those three classes so we make one check rather than three?

Reply:
actually ProviderAuthError alraedy `extends ProviderApiError`, so we can remove ProviderAuthError from the check. `ProviderUserError extends Error` but I guess we could make it extend ProviderApiError too

Reply:
If that's not too complicated, let's do that; otherwise, it's not a blocker, we can live with 2 `instanceof` checks